src/main/java/io/axoniq/demo/university/faculty/automation/allcoursesfullybookednotifier/WhenAllCoursesFullyBookedThenSendNotification.java

```diff
@@ -0,0 +1,148 @@
+            courses.computeIfPresent(event.courseId(), (id, course) -> course.capacity(event.capacity()));
+            return new State(courses, notified);
+        }
+
+        @EventSourcingHandler
+        State evolve(StudentSubscribedToCourse event) {
+            courses.computeIfPresent(event.courseId(), (id, course) -> course.studentSubscribed());
+            return new State(courses, notified);
+        }
+
+        @EventSourcingHandler
+        State evolve(StudentUnsubscribedFromCourse event) {
+            courses.computeIfPresent(event.courseId(), (id, course) -> course.studentUnsubscribed());
+            return new State(courses, notified);
+        }
+
+        @EventSourcingHandler
+        State evolve(AllCoursesFullyBookedNotificationSent event) {
+            return new State(courses, true);
+        }
+    }
+
+    static class AutomationCommandHandler {
+
+        @CommandHandler
+        public void decide(
+                SendAllCoursesFullyBookedNotification command,
+                @InjectEntity(idProperty = FacultyTags.FACULTY_ID) State state,
+                ProcessingContext context
+        ) {
+            var canNotify = state != null && !state.notified();
+            if (canNotify) {
+                var notification = new NotificationService.Notification("admin", "All courses are fully booked now.");
+                context.component(NotificationService.class).sendNotification(notification);
+                var eventAppender = EventAppender.forContext(context);
+                eventAppender.append(new AllCoursesFullyBookedNotificationSent(command.facultyId()));
+            }
+        }
+    }
+
+    static class AutomationEventHandler {
+
+        @EventHandler
+        public MessageStream.Empty<?> react(
+                StudentSubscribedToCourse event,
+                ProcessingContext context
+        ) {
+            var state = context.component(StateManager.class).loadEntity(State.class, FACULTY_ID, context).join();
+            return sendNotificationIfAllCoursesFullyBooked(state, context);
+        }
+
+        @EventHandler
+        public MessageStream.Empty<?> react(
+                CourseCapacityChanged event,
+                ProcessingContext context
+        ) {
+            var state = context.component(StateManager.class).loadEntity(State.class, FACULTY_ID, context).join();
+            return sendNotificationIfAllCoursesFullyBooked(state, context);
+        }
+
+        private MessageStream.Empty<Message> sendNotificationIfAllCoursesFullyBooked(State state, ProcessingContext context) {
+            var automationState = state != null ? state : new State();
+            var allCoursesFullyBooked = automationState.courses.values().stream().allMatch(State.Course::isFullyBooked);
+            var shouldNotify = allCoursesFullyBooked && !automationState.notified();
+            if (shouldNotify) {
+                var commandGateway = context.component(CommandGateway.class);
```

Comment:
Why not use the `CommandDispatcher`, which can be injected in the event handler annotated method? Worked on that before M3 to get it as part of automations like this :see_no_evil:

Reply:
No worries! Lots are going on. Tell me when you've made the adjustments! :-)